phpBB/phpbb/notification/type/base.php

```diff
@@ -286,13 +291,19 @@ public function prepare_for_display()
 
 		if ($this->get_url())
 		{
-			$u_mark_read = append_sid($this->phpbb_root_path . 'index.' . $this->php_ext, 'mark_notification=' . $this->notification_id . '&amp;hash=' . $mark_hash);
+			$query = http_build_query([
```

Comment:
Why are you using `http_build_query` and not just passing the parameter like this?:
```
$u_mark_read = $this->controller_helper->route('phpbb_notifications_mark_read', ['id' => $this->notification_id, 'hash' => $mark_hash]);
```